module-3/js-async.html

```diff
@@ -0,0 +1,87 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Fetch Operation</title>
+</head>
+<body>
+    <div id="btcprice"></div>
+    <div id="resultContainer"></div>
+
+    <script>
+
+    function fetchData() {
+        fetch('https://api.coingecko.com/api/v3/simple/price?ids=bitcoin&vs_currencies=usd')
+        .then(response => {
+            if (!response.ok) {
+                throw new Error('Network response was not ok');
+            }
+            return response.json();
+        })
+        .then(data => {
+            document.getElementById("btcprice").textContent = "BTC / USD: " + data.bitcoin.usd;
+        })
+    }
+
+    fetchData();
+
+    async function fakeAsyncOperation() {
+        return new Promise((resolve, reject) => {
+            setTimeout(() => {
+                const success = Math.random() < 0.8; 
+                if (success) {
+                    const data = { message: "Data fetched successfully." };
+                    resolve(data);
+                } else {
+                    reject(new Error("Failed to fetch data."));
+                }
+            }, 1000);
+        });
+    }
+
+        async function doAsyncThing() {
+            const resultContainer = document.getElementById('resultContainer');
+            resultContainer.textContent = "Loading..."; 
+            
+            try {
+                // Call fakeAsyncOperation using await
+                const data = await fakeAsyncOperation();
+                resultContainer.textContent = data.message;
+            } catch (error) {
+                resultContainer.textContent = "Error: " + error.message; 
+            }
+        }
+
+        async function doAsyncThing() {
+            const resultContainer = document.getElementById('resultContainer');
+            resultContainer.textContent = "Loading..."; 
+            
+            try {
+                // Call fakeAsyncOperation using await
+                const data = await fakeAsyncOperation();
+                resultContainer.textContent = data.message; // Display success message
+            } catch (error) {
+                resultContainer.textContent = "Error: " + error.message; // Display error message
+            }
+        }
+
+        async function doAsyncThing2(asyncFunction) {
+            const resultContainer = document.getElementById('resultContainer');
+            resultContainer.textContent = "Loading..."; // Display loading message
+            
+            try {
+                const data = await asyncFunction();
+                resultContainer.textContent = data.message; 
+            } catch (error) {
+                resultContainer.textContent = "Error: " + error.message; 
+            }
+        }
+
+        doAsyncThing2()
```

Comment:
Here, this `doAsyncThing2()` function is called without arguments; but the function signature requires one argument.

<img width="270" alt="image" src="https://github.com/ym-xu/metana-bootcamp/assets/394065/f86d6e39-3554-4f96-9677-72956494ee32">